Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.38252&attempt_id=88e6ece9-65ec-4158-8245-12173f9681cf)

Attempted: ted

<details><summary>Partial log (click to expand)</summary><p>

```
clang -g -O2 -I/nix/store/97sp2fls12j29fsfznpjqnh8max5v4wy-freetype-2.7.1-dev/include/freetype2 -I/nix/store/97sp2fls12j29fsfznpjqnh8max5v4wy-freetype-2.7.1-dev/include/freetype2 -I/nix/store/4qmrqhbnbh6nrqj95ymd8qxnrs6gcj4r-xproto-7.0.31/include -I/nix/store/127spfiigv502gv6q0pjdviir219fk01-libXft-2.3.2-dev/include  -I/nix/store/4qmrqhbnbh6nrqj95ymd8qxnrs6gcj4r-xproto-7.0.31/include -I/nix/store/38nimfc0r7vx2vy2nqyibli01ljpv9yf-libX11-1.6.5-dev/include -I/nix/store/96p8r2c4vmii4r4iyx2dva1k4hxw2ir0-kbproto-1.0.7/include -I/nix/store/4gxliil1zqpqidgdspg8rygvqp8cg7zs-libXpm-3.5.12-dev/include -I/nix/store/gngg6fi3jgijrq0v3z0a6n8p1r109z8r-gtk+-2.24.32-dev/include/gtk-2.0 -I/nix/store/q2msz613i3xi168gh3yn9ac5bpp5vs8h-gtk+-2.24.32/lib/gtk-2.0/include -I/nix/store/k1zk26s88yi776639dll9a3n060gdc8c-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/srlcwaq125gpdr4wd7my73gw0xqad61r-glib-2.56.0/lib/glib-2.0/include -I/nix/store/95wq83h3yz5x8w5a7902cclh49hd1cb9-cairo-1.14.10-dev/include/cairo -I/nix/store/97sp2fls12j29fsfznpjqnh8max5v4wy-freetype-2.7.1-dev/include/freetype2 -I/nix/store/97sp2fls12j29fsfznpjqnh8max5v4wy-freetype-2.7.1-dev/include -I/nix/store/d7dznzs843lmdac4ns1qvl3ykf868i4s-pango-1.40.14-dev/include/pango-1.0 -I/nix/store/5bi7lrd2cmwva1fkk7xc3psjp5x1bv9i-gdk-pixbuf-2.36.7-dev/include/gdk-pixbuf-2.0 -I/nix/store/r4vb7b9kj829bgb8y2nwrn63kxq1im35-atk-2.28.1-dev/include/atk-1.0 -I../guiInterface -I../appFrame -I../drawMeta -I../appUtil -I../textEncoding -I../utilPs -I../docFont -I../bitmap -I../ind -I../include      -c -o drawDrawingSurfaceGtk.o drawDrawingSurfaceGtk.c
drawDrawingSurfaceGtk.c:429:13: fatal error: 'gdk/gdkx.h' file not found
#   include     <gdk/gdkx.h>
                ^~~~~~~~~~~~
1 error generated.
make[1]: *** [<builtin>: drawDrawingSurfaceGtk.o] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-ted-2.23.drv-0/Ted-2.23/appFrame'
make: *** [Makefile:212: lib/appFrame.a] Error 2
builder for '/nix/store/lxryl2r8lcxxcac2107y3bh0xkqwfkgc-ted-2.23.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/lxryl2r8lcxxcac2107y3bh0xkqwfkgc-ted-2.23.drv' failed
```
</p></details>

--- a/pkgs/tools/typesetting/ted/default.nix
+++ b/pkgs/tools/typesetting/ted/default.nix
@@ -1,0 +1,84 @@
+{ stdenv, fetchurl, pkgconfig, zlib, pcre, xorg, libjpeg, libtiff, libpng, gtk2, libpaper, makeWrapper, ghostscript }:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "ted";
+  version = "2.23";
+
+  src = fetchurl {
+    url = "http://ftp.nluug.nl/pub/editors/${pname}/${name}.src.tar.gz";
+    sha256 = "0v1ipynyjklb3chd1vq26a21sjjg66sir57gi2kkrbwnpk195a9z";
+  };
+
+  preConfigure = ''
+    mkdir pkgconfig-append
+    pushd pkgconfig-append
+
+    # ted looks for libtiff, not libtiff-4 in its pkg-config invokations
+    cp ${libtiff.dev}/lib/pkgconfig/libtiff-4.pc libtiff.pc
+
+    # ted needs a libpaper pkg-config file
+    cat > libpaper.pc << EOF
+    prefix=${libpaper}
+    libdir=${libpaper}/lib
+    includedir=${libpaper}/include
+    exec_prefix=\''${prefix}
+
+    Name: libpaper
+    Version: ${libpaper.version}
+    Description: ${libpaper.meta.description}
+    Libs: -L\''${libdir} -lpaper
+    Cflags: -I\''${includedir}
+    EOF
+
+    export PKG_CONFIG_PATH="$PWD:$PKG_CONFIG_PATH"
+
+    popd
+  '';
+
+  makeFlags = [ "CONFIGURE_OPTIONS=--with-GTK" "CONFIGURE_OPTIONS+=--prefix=$(out)" "compile.shared" ];
+
+  installPhase = ''
+    runHook preInstall
+
+    make tedPackage/makefile
+    pushd tedPackage
+    substituteInPlace makefile --replace /usr ""
+    make PKGDESTDIR=$out datadir
+    popd
+
+    pushd $out/share/Ted/examples
+    for f in rtf2*.sh
+    do
+        makeWrapper "$PWD/$f" "$out/bin/$f" --prefix PATH : $out/bin:${stdenv.lib.makeBinPath [ ghostscript ]}
+    done
+    popd
+
+    cp -v Ted/Ted $out/bin
+
+    runHook postInstall
+  '';
+
+  buildInputs = [ pkgconfig zlib pcre xorg.xlibsWrapper xorg.libXpm libjpeg libtiff libpng gtk2 libpaper makeWrapper ];
+
+  meta = with stdenv.lib; {
+    description = "Ted, an easy rich text processor";
+    longDescription = ''
+      Ted is a text processor running under X Windows on Unix/Linux systems.
+      Ted was developed as a standard easy light weight word processor, having
+      the role of Wordpad on MS-Windows. Since then, Ted has evolved to a real
+      word processor. It still has the same easy appearance and the same speed
+      as the original. The possibility to type a letter, a note or a report
+      with a simple light weight program on a Unix/Linux machine is clearly
+      missing. Ted was made to make it possible to edit rich text documents on
+      Unix/Linux in a wysiwyg way. RTF files from Ted are fully compatible with
+      MS-Word. Additionally, Ted also is an RTF to PostScript and an RTF to
+      Acrobat PDF converter.
+    '';
+    homepage    = https://nllgg.nl/Ted/;
+    license     = licenses.gpl2;
+    platforms   = platforms.all;
+    broken      = stdenv.isDarwin;
+    maintainers = with maintainers; [ obadz ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5023,6 +5023,8 @@ with pkgs;
 
   tcpreplay = callPackage ../tools/networking/tcpreplay { };
 
+  ted = callPackage ../tools/typesetting/ted { };
+
   teamviewer = callPackage ../applications/networking/remote/teamviewer {
     stdenv = stdenv_32bit;
   };

